app/components/chart-stacked-bar.js

```diff
@@ -0,0 +1,97 @@
+import Ember from 'ember';
+
+import { select } from 'd3-selection';
+import { scaleOrdinal, scaleBand, scaleLinear, schemeCategory10 } from 'd3-scale';
+import { max } from 'd3-array';
+import { axisBottom, axisLeft } from 'd3-axis';
+import { stack } from 'd3-shape';
+
+const { Component, run, get } = Ember;
+
+export default Component.extend({
+  tagName: 'svg',
+  classNames: ['chart-bar-stacked'],
+  attributeBindings: ['width', 'height'],
+  didReceiveAttrs() {
+    // Anytime we get an update schedule a draw
+    run.scheduleOnce('render', this, this.draw);
+  },
+  data: null,
+  width: null,
+  height: null,
+  draw(){
+    const data = get(this, 'data') || [];
+    const svg = select(this.element);
+    const margin = {top: 0, right: 20, bottom: 40, left: 20};
+    const chartWidth = get(this, 'width') - margin.left - margin.right;
+    const chartHeight = get(this, 'height') - margin.top - margin.bottom;
+
+    const x = scaleBand().range([0, chartWidth]).padding(0.4);
+    const y = scaleLinear().range([chartHeight, 0]);
+    const z = scaleOrdinal(schemeCategory10);
+
+    var keys_set = {};
+    var keys_list = [];
+
+    let reformattedData = data.map((d) => {
+      let obj = {label: d.label, total: d.total};
+      d.values.forEach((value, index) => {
+        var value_key = value.label + '-' + index;
+        obj[value_key] = value.value;
+        if (!keys_set[value_key]) {
+          keys_set[value_key] = 1;
+          keys_list.push(value_key);
+        }
+      });
+      return obj;
+    }).map((d) => {
+      keys_list.forEach((key) => {
+        d[key] = d[key] || 0;
+      });
+      return d;
+    });
+
+    const dataOrArray = reformattedData?reformattedData:[];
```

Comment:
Sometimes data will be null which is going to blow up the `data.map` above.  This dataOrArray call should still go in front of the `reformattedData` section.